### DISCUSSION TITLE
Unable to connect to simulator?

### DISCUSSION THREAD

**Discussion by c00lzerosoftware** [everysight-maverick/sdk#1] [Q&A] [answered]
Pretty empty here, anyone messing around with this sdk?  I'm having trouble getting connect to the simulator.  Granted I barely know what I'm doing, but this builds fine and run stops "No target device found."

`
package com.everysight.sdk.samples.helloworld

import UIKit.services.AppErrorCode
import UIKit.services.IEvsAppEvents
import UIKit.services.IEvsCommunicationEvents
import android.Manifest
import android.app.Activity
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import com.everysight.evskit.android.Evs

class HelloWorldActivity : Activity(), IEvsCommunicationEvents, IEvsAppEvents {

    private val screen = HelloWorldScreen()
    private lateinit var txtStatus: TextView
    companion object{
        private const val TAG = "HelloWorldActivity"
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        //setup the sdk
        val options = hashSetOf("supportSimulator")
        Evs.init(this).startExt(options)

        setContentView(R.layout.activity_hello_world)

        // Set up communication with the Maverick Simulator
        Evs.instance().comm().setDeviceInfo("udp://198.168.50.134", "Simulator")

        checkPerm()

        initSdk()

        Evs.instance().screens().addScreen(screen)
        txtStatus = findViewById(R.id.txtStatus)

        findViewById<Button>(R.id.btnConfigure).setOnClickListener{
            Evs.instance().showUI("configure")
        }
        findViewById<Button>(R.id.btnSettings).setOnClickListener{
            Evs.instance().showUI("settings")
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        Evs.instance().unregisterAppEvents(this)
        Evs.instance().comm().unregisterCommunicationEvents(this)
        Evs.instance().stop()
    }

    private fun initSdk() {
        Evs.init(this).start()
        Evs.startDefaultLogger()//optional
        Evs.instance().registerAppEvents(this)
        with(Evs.instance().comm()){
            registerCommunicationEvents(this@HelloWorldActivity)
            if(hasConfiguredDevice()) connect()
        }

    }

    private fun checkPerm() {
        var permissionsRequested = ArrayList<String>()
        permissionsRequested.add(Manifest.permission.ACCESS_COARSE_LOCATION)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
            permissionsRequested.add(Manifest.permission.BLUETOOTH_SCAN)
            permissionsRequested.add(Manifest.permission.BLUETOOTH_CONNECT)
        }else{
            permissionsRequested.add(Manifest.permission.BLUETOOTH)
        }
        permissionsRequested = validatePermissions(permissionsRequested)
        if (permissionsRequested.size > 0) {

            Log.d(TAG,"validating permissions")
            val req = permissionsRequested.toTypedArray()
            requestPermissions(req, 666)
        }
    }
    private fun validatePermissions(permissionsRequested: ArrayList<String>): ArrayList<String> {
        val permissionsToRequest = ArrayList<String>()
        for (permission in permissionsRequested) {
            if (checkSelfPermission(permission) != PackageManager.PERMISSION_GRANTED) {
                permissionsToRequest.add(permission)
            }
        }
        return permissionsToRequest
    }

    override fun onAdapterStateChanged(isEnabled: Boolean) {
        runOnUiThread{txtStatus.text = "Adapter enabled=$isEnabled"}
    }

    override fun onConnected() {
        runOnUiThread{txtStatus.text = "${Evs.instance().comm().getDeviceName()} is Connected"}
    }

    override fun onConnecting() {
        runOnUiThread{txtStatus.text = "Connecting ${Evs.instance().comm().getDeviceName()}"}
    }

    override fun onDisconnected() {
        runOnUiThread{txtStatus.text = "Disconnected"}
    }

    override fun onFailedToConnect() {
        runOnUiThread{txtStatus.text = "${Evs.instance().comm().getDeviceName()} Failed to connect"}
    }

    override fun onReady() {
        Evs.instance().display().turnDisplayOn()
    }

    override fun onError(errCode: AppErrorCode, description: String) {
        runOnUiThread{txtStatus.text = "Error: $errCode"}
    }
}
`

![image](https://github.com/everysight-maverick/sdk/assets/95159895/a6cc2f11-2c02-4e65-bec8-992d1d783422)

**Comment by feranf** [comment, marked as answer]
Hi,

The simulator simulates the glasses display only and not an Android  phone. The application is installed on the phone and connected the simulator (via WiFi) - PC and Phone must be on the same network.

The "No Target Device Found" is a standard message from the Android Studio states that the Android Studio cant find a target Android phone to install the "hello world" application on it.

Please make sure your phone is connected to your PC and the USB debugging is enabled via the developers options.
Please refer the [https://developer.android.com/codelabs/basic-android-kotlin-compose-connect-device#0](https://developer.android.com/codelabs/basic-android-kotlin-compose-connect-device#0) for more information.

Hope that helped :-)

**Reply by c00lzerosoftware** [reply]
Ah, got it, apologies I thought the simulator did everything, this worked thank you!

**Reply by TitusCF** [reply]
Hi! I just got my Maverick today, have you been able to make any progress with yours?

**Reply by c00lzerosoftware** [reply]
Hi Titus, yes I got all examples and one of my own test apps to the simulator but wanted to wait for the hardware to continue, so I took a break.  I just got my hardware and have run all on it just fine.  I am new at Android coding, so for me it was a bit tough getting the samples up but I eventually figured it out and the rest went smoother once I knew what I needed to do. The samples as given should find Maverick and run...just make sure to close the app once done so it finds and connects to the next you want to run. I'll start checking this git again for updates if you'd like some help.  ChatGPT and feranf helped me get it going lol.
